### ISSUE TITLE
aws_datasync_task: Cloud watch log group ARN regex validation

### ISSUE THREAD

**Issue by stellemd** [terraform-providers/terraform-provider-aws#13046]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform version: 0.12.24  

### Affected Resource(s)

* aws_datasync_task

### Terraform Configuration Files
```hcl
resource "aws_datasync_task" "this" {
  destination_location_arn = "arn:aws:datasync:us-east-1:123456789123:location/loc-02ca0734e27104g8d"
  source_location_arn      = "arn:aws:datasync:us-east-1:123456789123:location/loc-Xcce5febb7a3a6a45"
  cloudwatch_log_group_arn = aws_cloudwatch_log_group.default.arn
  name                     = "Test Media Sync"

  options {
          atime                  = "BEST_EFFORT"
          bytes_per_second       = -1
          gid                    = "INT_VALUE"
          mtime                  = "PRESERVE"
          posix_permissions      = "PRESERVE"
          preserve_deleted_files = "REMOVE"
          preserve_devices       = "NONE"
          uid                    = "INT_VALUE"
          verify_mode            = "POINT_IN_TIME_CONSISTENT"
        }
}

resource "aws_cloudwatch_log_group" "default" {
  name              = local.cloudwatch_log_group_name
  retention_in_days = 400

  tags =  local.cloudwatch_log_group_tags
}
```

### Debug Output

https://github.com/stellemd/aws_data_sync_task/blob/master/debug.txt

### Expected Behavior

The aws data sync task resource will accept the cloud watch log group arn created via an aws_cloudwatch_log_group resource and the cloud watch log group will be attached for task logging.

### Actual Behavior
```hcl
Error: error creating DataSync Task: ValidationException: 1 validation error detected: Value 'arn:aws:logs:us-east-1:123456789123:log-group:test_media_data_sync_non_prod:*' at 'cloudWatchLogGroupArn' failed to satisfy constraint: Member must satisfy regular expression pattern: ^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b):logs:[a-z\-0-9]*:[0-9]{12}:log-group:([^:\*]*)$
        status code: 400, request id: 2883fee4-5bf5-4969-8428-bb90fe6fc23b

  on main.tf line 35, in resource "aws_datasync_task" "this":
  35: resource "aws_datasync_task" "this" {
```
### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by mehfuzh** [comment]
Actually, this is a major issue

**Comment by bflad** [comment]
FYI, this should be resolved in our upcoming major release (version 3.0.0) within the next few weeks. We plan on automatically removing the `:*` returned in the ARN by the CloudWatch Logs API, so it is like all other Terraform AWS Provider `arn` attributes. I have marked this issue with that milestone so we know to circle back and close this issue when its resolved as part of that release.

**Comment by bflad** [comment]
Hi folks 👋  The `aws_cloudwatch_log_group` change mentioned above has been merged and will go out with version 3.0.0 of the Terraform AWS Provider, likely later today or tomorrow. Configurations in the meantime and with earlier versions of the provider can use the `replace()` function to remove the suffix. 👍

**Comment by MeMan-MasterOfTheUniverse** [comment]
That is not the ARN. You're changing a value that doesn't originate within terraform!?
